Option for redeem violations

yulgolem opened the pull request:
Closes #590

vohmar commented:
after a users got third ban problems started with being able to log into the portal "You need to sign in or sign up before continuing." Other users seem not to be affected. 

logs do not reveal much
successful : https://gist.github.com/ratM1n/0f53b2d3b169acbb7a5894d9af42fb40 
unsuccessful: https://gist.github.com/ratM1n/f18338f929035c09cc1fbe888037cd97

from time to time the login is successful, but mostly not. 

After deleting one ban from db, was able to log in instantly.

now after paying for one of the overdue invoices (not the last one) I still have the notification of hard ban in the interface and I am not able to make any bids. 

paid for the last violation as well - still no ability to participate in auctions and still have the red notification in the ui.

After logging out and then logging in again got the correct notification of 1 active violation.

Have one concern - the 3rd violation record does not have reference of the domain name the record is related to - I think that in the case where its possible to redeem the violation by paying any overdue invoice we should change that feature, have domain name associated to all violations and we use just the count of active violations to indicate between soft and hard ban.

yulgolem commented:
@vohmar @OlegPhenomenon I suppose there might be some issues on Tara side with logging in by MobileId/telefoninumber.
Added a system test to be sure on our side all OK with Tara and invoices are payable.

Added unpaid invoices, two bans on invoices and one admin UI ban to Tara test user with Smart-ID code: 10101010005. We still can log in as this user and invoices got payment options.

yulgolem commented:
1) all bans should be long (about 1 year - from Setting.find_by(code: 'ban_length'), not 3 months from  SHORT_BAN_PERIOD_IN_MONTHS).
1) all bans should have domain name/invoice included.
1) any count of bans assigned should allow user to log in and pay canceled invoice, but 3+ bans restricts user's ability to see auctions etc.
1) 3+active  bans restriction is calculated on a user-based method, not by presence of ban without domain name.
1) paying invoice removes related ban.
